File: java/liskov-plus-refactor/src/main/java/no/bouvet/solid/lsp/services/RefundService.java
package no.bouvet.solid.lsp.services;

import java.math.BigDecimal;

import no.bouvet.solid.lsp.domain.Payment;

public class RefundService {

    public boolean refund(PaymentType paymentServiceType, BigDecimal amount, String transactionId) {

        Payment payment = PaymentFactory.getPayment(paymentServiceType);
        return payment.refund(amount, transactionId);

    }

}

File: dotnet/Solid/RefactoredSolid/LSP/Example2/Services/RefundService.cs
using RefactoredSolid.LSP.Example2.Domain;

namespace RefactoredSolid.LSP.Example2.Services
{
    public class RefundService
    {
        public bool Refund(PaymentType paymentServiceType, decimal amount, string transactionId)
        {
            Payment payment = PaymentFactory.GetPayment(paymentServiceType);
            return payment.Refund(amount, transactionId);
        }
    }
}
